### PR TITLE
Transition show / hide for toggling controls

### PR DESCRIPTION
**Pull request by cathyxz**
I used animation instead of css transition here because css transition doesn't play nice with `display: none`, and we want to use `display: none` to make sure that we don't accidentally click on the button. The extra `keyframe` classes are for backwards compatibility. I've seen them elsewhere in the codebase, but I don't know if for example our build process handles them already.

### CODE REVIEW

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css [hunk 1]
```diff
@@ -212,3 +228,39 @@
   bottom: 0;
   right: 0;
 }
+
+@-webkit-keyframes fadeIn {
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: same with @keyframme

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css [hunk 2]
```diff
@@ -55,10 +55,24 @@
   top: 0 !important;
   height: 50px !important;
   background: linear-gradient(rgba(0,0,0,0.3), rgba(0,0,0,0));
-  transition: opacity 1s;
   z-index: 2;
 }
 
+.i-amphtml-lbv-controls {
+  opacity: 0;
+  -webkit-animation: fadeIn ease-in 0.4s 1 forwards;
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: no need for prefixing, our CSS compiler will do that

Comment (cathyxz) [reply extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: Cool! removed.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css (hunk 1 above)
Comment (cathyxz) [reply extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: Ditto here.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css [hunk 3]
```diff
@@ -55,10 +55,20 @@
...
 }
 
+.i-amphtml-lbv-controls {
+  opacity: 0;
+  animation: fadeIn ease-in 0.4s 1 forwards;
+}
+
+.i-amphtml-lbv-controls.fade-out
+ {
+  opacity: 0;
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: the opacity should not be needed here right? (or should be 1 to match the animation)

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css [hunk 4]
```diff
@@ -75,9 +85,11 @@
   bottom: 0 !important;
   overflow-x: hidden !important;
   color: #ffffff;
-  opacity: 1;
-  transition: opacity 1s;
   z-index: 2;
+  opacity: 0;
+  -webkit-animation: fadeIn ease-in 0.4s 1 forwards;
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: remove prefix

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css [hunk 5]
```diff
@@ -212,3 +224,16 @@
   bottom: 0;
   right: 0;
 }
+
+@keyframes fadeIn {
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.css]: for both fadeIn and fadeOut, `opacity` alone is not enough to hide (it keeps the invisible item tappable). They need `visibility:hidden/visible` as well.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 6]
```diff
@@ -504,12 +507,12 @@ export class AmpLightboxViewer extends AMP.BaseElement {
 
   /**
    * Toggle description box if it has text content
-   * @param {boolean} display
    * @private
    */
-  toggleDescriptionBox_(display) {
+  showDescriptionBox_() {
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: let's remove this and just do it inline in the `if` condition like the hide case.

File: extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js [hunk 7]
```diff
@@ -504,12 +507,12 @@ export class AmpLightboxViewer extends AMP.BaseElement {
...
+    this.descriptionBox_.classList.remove('fade-out');
     this.updateDescriptionBox_();
-    toggle(dev().assertElement(this.descriptionBox_), display);
+    toggle(dev().assertElement(this.descriptionBox_), true);
```
Comment (aghassemi) [inline extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js]: there should not be a need to keep `toggle` given opacity/visibility in CSS will handle it.

**Pull request merged**